Meta/Lagom/Tools/CodeGenerators/LibWeb/BindingsGenerator/IDLGenerators.cpp

```diff
@@ -3754,44 +3755,157 @@ JS_DEFINE_NATIVE_FUNCTION(@class_name@::@attribute.getter_callback@)
                 //    1. Let parsedValue be the result of integer parsing contentAttributeValue if the reflected IDL attribute is not limited to only non-negative numbers;
                 //       otherwise the result of non-negative integer parsing contentAttributeValue.
                 //    2. If parsedValue is not an error and is within the long range, then return parsedValue.
+                auto positive = false;
+
+                if (attribute.extended_attributes.contains("ReflectNonNegative"))
+                    positive = true;
+
+                attribute_generator.set("positive", positive ? "true"sv : "false"sv);
+
+                // 3. If the reflected IDL attribute has a default value, then return defaultValue.
+                if (attribute.extended_attributes.contains("ReflectFallback")) {
+                    auto default_value = attribute.extended_attributes.get("ReflectFallback").value();
+                    attribute_generator.set("has_default_value", "true"sv);
+                    attribute_generator.set("default_value", default_value);
+                } else {
+                    attribute_generator.set("has_default_value", "false"sv);
+                    attribute_generator.set("default_value", "0"sv);
+                }
+
+                // 4. If the reflected IDL attribute is limited to only non-negative numbers, then return −1.
+                // 5. Return 0.
                 attribute_generator.append(R"~~~(
     i32 retval = 0;
+    bool did_set_value = false;
     auto content_attribute_value = impl->get_attribute("@attribute.reflect_name@"_fly_string);
     if (content_attribute_value.has_value()) {
-        auto maybe_parsed_value = Web::HTML::parse_integer(*content_attribute_value);
-        if (maybe_parsed_value.has_value())
-            retval = *maybe_parsed_value;
+        if (@positive@) {
+            auto maybe_parsed_value = Web::HTML::parse_non_negative_integer(*content_attribute_value);
+
+            if (maybe_parsed_value.has_value()) {
+                auto parsed_value = maybe_parsed_value.value();
+
+                if (parsed_value <= NumericLimits<WebIDL::Long>::max()) {
```

Comment:
When can the resulting value of `parse_non_negative_integer` (`Optional<u32>`) be larger than `NumericLimits<WebIDL::Long>::max()`?